File: PolicyManagementAngular/src/app/app-entites/models/common/search-policy-model.ts
import { PolicyManagement } from "src/app/shared/utilities/enums/enum";

export interface ISearchPolicyModel {
    ControlNumber: IControlNumber;
    CustomerName: string;
    InsuranceCompany: number;
    PolicyNumber: string;
    RegistrationNumber: string;
    Manufacture: number;
    Model: number;
    Pos: number;
    PolicyStartDateFrom: string;
    PolicyStartDateTo: string;
    PolicyEndDateFrom: string;
    PolicyEndDateTo: string;
    MobileNumber: string;
    Vertical: number;
    Product: number;
    PolicyManagementType: PolicyManagement;
    // PageNumber: number;
    // PageSize: number;
    IsForDownload : boolean;
    IsForShowAll : boolean; 
}

export interface IControlNumber {
    Year: string;
    BranchCode: string;
    VerticalCode: string;
    Number: string;
}

File: PolicyManagement.Api/PolicyManagement.Models/Common/SearchPolicyModel.cs
using PolicyManagement.Utilities.Enums;

namespace PolicyManagement.Models.Common
{
    public class SearchPolicyModel : BaseModel
    {
        public ControlNumber ControlNumber { get; set; }
        public string CustomerName { get; set; }
        public int InsuranceCompany { get; set; }
        public string PolicyNumber { get; set; }
        public string RegistrationNumber { get; set; }
        public int Manufacture { get; set; }
        public int Model { get; set; }
        public int Pos { get; set; }
        public string PolicyStartDateFrom { get; set; }
        public string PolicyStartDateTo { get; set; }
        public string PolicyEndDateFrom { get; set; }
        public string PolicyEndDateTo { get; set; }
        public string MobileNumber { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public PolicyManagementType PolicyManagementType { get; set; }
        public int Product { get; set; }
        public int Vertical { get; set; } 
        public bool IsForDownload { get; set; } 
        public bool IsForShowAll { get; set; } 
    }

    public class ControlNumber
    {
        public string Year { get; set; }
        public string BranchCode { get; set; }
        public string VerticalCode { get; set; }
        public string Number { get; set; }
    }
}
